commit 5d0255e8045a36a1636606c60c7be234a1ef2436
Author: David Lin <dtwlin@google.com>
Date:   Tue May 23 21:53:55 2017 -0400

    jump label: pass kbuild_cflags when checking for asm goto support
    
    
    [ Upstream commit 35f860f9ba6aac56cc38e8b18916d833a83f1157 ]
    
    Some versions of ARM GCC compiler such as Android toolchain throws in a
    '-fpic' flag by default.  This causes the gcc-goto check script to fail
    although some config would have '-fno-pic' flag in the KBUILD_CFLAGS.
    
    This patch passes the KBUILD_CFLAGS to the check script so that the
    script does not rely on the default config from different compilers.
    
    Link: http://lkml.kernel.org/r/20170120234329.78868-1-dtwlin@google.com
    Signed-off-by: David Lin <dtwlin@google.com>
    Acked-by: Steven Rostedt <rostedt@goodmis.org>
    Cc: Michal Marek <mmarek@suse.com>
    Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
    Signed-off-by: Linus Torvalds <torvalds@linux-foundation.org>
    Signed-off-by: Sasha Levin <alexander.levin@verizon.com>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>

diff --git a/Makefile b/Makefile
index 8a49706..c93e394 100644
--- a/Makefile
+++ b/Makefile
@@ -801,7 +801,7 @@ KBUILD_CFLAGS   += $(call cc-option,-Werror=incompatible-pointer-types)
 KBUILD_ARFLAGS := $(call ar-option,D)
 
 # check for 'asm goto'
-ifeq ($(shell $(CONFIG_SHELL) $(srctree)/scripts/gcc-goto.sh $(CC)), y)
+ifeq ($(shell $(CONFIG_SHELL) $(srctree)/scripts/gcc-goto.sh $(CC) $(KBUILD_CFLAGS)), y)
 	KBUILD_CFLAGS += -DCC_HAVE_ASM_GOTO
 	KBUILD_AFLAGS += -DCC_HAVE_ASM_GOTO
 endif
